.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check_duplicate_runs:
+    name: Check for duplicate runs
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          concurrent_skipping: always
+          cancel_others: true
+          skip_after_successful_duplicate: true
+          paths_ignore: '["**/README.md", "**/CHANGELOG.md", "**/LICENSE"]'
+          do_not_skip: '["pull_request"]'
+
+  tests:
+    name: Run tests (Elixir ${{matrix.elixir}}, OTP ${{matrix.otp}})
+    needs: check_duplicate_runs
+    if: ${{ needs.check_duplicate_runs.outputs.should_skip != 'true' }}
+
+    strategy:
+      matrix:
+        elixir:
+        - "1.7"
+        - "1.8"
+        - "1.9"
+        - "1.10"
+        - "1.11"
+        otp:
+        - "20.3"
+        - "21.3"
+        - "22.3"
+        - "23.3"
+        exclude:
+        - elixir: "1.7"
+          otp: "23.3"
+        - elixir: "1.8"
+          otp: "23.3"
+        - elixir: "1.9"
+          otp: "23.3"
+        - elixir: "1.10"
+          otp: "20.3"
+        - elixir: "1.11"
+          otp: "20.3"
+        - elixir: "1.11"
+          otp: "21.0"
```

Comment:
I don't think we need such wide build matrix for this particular project. I like to build on just the oldest supported and newest available Elixir/OTP combo and that tends to be good enough to find compatibility bugs.

Reply:
Fixed.